### PR TITLE
[SPARK-6305][BUILD][FOLLOWUP] Some cleanups for log4j properties

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

This is a followup of SPARK-6305. Some places in build files (maven and sbt) which remove log4j properties files should be updated. Some tests programmingly write log4j properties should be updated to log4j 2 syntax too. 

### Why are the changes needed?

To clean up build files and migrate some tests to log4j2 syntax.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass all tests.

### CODE REVIEW

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (if CI passed).

**Comment by viirya** [comment]
Thanks @dongjoon-hyun . Wait for CI.

**Comment by attilapiros** [comment]
I think there is a related error in the kubernetes integration tests:

```
  <testcase
  name="Verify logging configuration is picked from the provided SPARK_CONF_DIR/log4j.properties" classname="org.apache.spark.deploy.k8s.integrationtest.KubernetesSuite" time="204.286">
    <failure
    message="The code passed to eventually never returned normally. Attempted 180 times over 3.000611025833334 minutes. Last failure message: &quot;++ id -u&amp;#010;+ myuid=185&amp;#010;++ id -g&amp;#010;+ mygid=0&amp;#01
0;+ set +e&amp;#010;++ getent passwd 185&amp;#010;+ uidentry=&amp;#010;+ set -e&amp;#010;+ '[' -z '' ']'&amp;#010;+ ...
```

It is around here:
https://github.com/apache/spark/blob/14a9d844e56afddeadf4d40fdf8f43c9fa3288fc/resource-managers/kubernetes/integration-tests/src/test/scala/org/apache/spark/deploy/k8s/integrationtest/SparkConfPropagateSuite.scala#L30-L45

Switching on DEBUG for log4j initialization does not trigger the logging of the expected output.

**Comment by viirya** [comment]
Hmm, I changed the log4j properties file for the k8s integration test. But seems the test still cannot get the log4j 2 configuration file. Not sure why now. I will merge this first and keep looking at the k8s integration test issue.

**Comment by viirya** [comment]
Jenkins tests were passed. Thanks for reviewing. Merging to master.

**Comment by LuciferYang** [comment]
@viirya It seems that test failure still exists in `hive-thriftserver` module:

https://amplab.cs.berkeley.edu/jenkins/job/spark-master-test-maven-hadoop-3.2/3527/consoleFull

**Comment by viirya** [comment]
Which one you mean?

**Comment by LuciferYang** [comment]
```
=========================================
End HiveThriftServer2Suite failure output
=========================================
       
22:53:44.821 INFO org.apache.spark.sql.hive.thriftserver.HiveThriftBinaryServerSuite: HiveThriftServer2 stopped
22:53:44.827 WARN org.apache.spark.sql.hive.thriftserver.HiveThriftBinaryServerSuite: 

===== POSSIBLE THREAD LEAK IN SUITE o.a.s.sql.hive.thriftserver.HiveThriftBinaryServerSuite, threads: Thread-13 (daemon=true), Thread-12 (daemon=true) =====

*** RUN ABORTED ***
  java.util.concurrent.TimeoutException: Futures timed out after [3 minutes]
  at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:259)
  at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:263)
  at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:293)
  at org.apache.spark.sql.hive.thriftserver.HiveThriftServer2TestBase.startThriftServer(HiveThriftServer2Suites.scala:1344)
  at org.apache.spark.sql.hive.thriftserver.HiveThriftServer2TestBase.$anonfun$beforeAll$4(HiveThriftServer2Suites.scala:1397)
  at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
  at scala.util.Try$.apply(Try.scala:213)
  at org.apache.spark.sql.hive.thriftserver.HiveThriftServer2TestBase.$anonfun$beforeAll$3(HiveThriftServer2Suites.scala:1396)
  at scala.util.Failure.orElse(Try.scala:224)
  at org.apache.spark.sql.hive.thriftserver.HiveThriftServer2TestBase.$anonfun$beforeAll$2(HiveThriftServer2Suites.scala:1394)
```

**Comment by viirya** [comment]
I will try to reproduce it locally.

**Comment by LuciferYang** [comment]
And there are some error log as follows:

```
Discovery starting.
2021-12-20 22:41:21,477 ScalaTest-main ERROR Filters contains invalid attributes "onMatch", "onMismatch"
2021-12-20 22:41:21,492 ScalaTest-main ERROR Filters contains invalid attributes "onMatch", "onMismatch"
Discovery completed in 1 second, 557 milliseconds.
Run starting. Expected test count is: 551
```

It seems that there is some thing wrong with log4j2 configuration.

**Comment by viirya** [comment]
The log4j error I have seen locally, but it doesn't affect SBT test. Not sure if it is related.

**Comment by LuciferYang** [comment]
> The log4j error I have seen locally, but it doesn't affect SBT test. Not sure if it is related.

ok

**Comment by viirya** [comment]
Do you see other issues? Seems other tests are all passed on maven, right?

**Comment by LuciferYang** [comment]
> Do you see other issues? Seems other tests are all passed on maven, right?

It seems so, only `hive-thriftserver` module test fail now

File: sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala
Comment (LuciferYang) [inline sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala:1231]: new File(s"$tempLog4jConf/log4j.properties") -> new File(s"$tempLog4jConf/log4j2.properties")

Comment (LuciferYang) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala:1231]: However, it seems that all logs are printed to the console, which is different from the previous behavior

Comment (viirya) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala:1231]: SBT cannot reproduce it. But we should use `log4j2.properties`.

Comment (viirya) [reply sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala:1231]: I corrected this together in #34965.

**Pull request closed without merging**
